src/compiler/checker.ts

```diff
@@ -981,6 +982,29 @@ namespace ts {
             }
         }
 
+        function checkAndReportErrorForUsingTypeAsValue(errorLocation: Node, name: string, meaning: SymbolFlags, nameNotFoundMessage: DiagnosticMessage, nameArg: string | Identifier): boolean {
+            const strictlyValueMeanings = SymbolFlags.Value & ~SymbolFlags.Type;

```

Comment:
I don't quite follow, sorry. I'll try to explain - the first filter is needed because the error message is specific to that meaning, so we just want to bail if you're not trying to resolve a value meaning. The second filter is needed because we don't want to show the message unless the symbol can be resolved as a type specifically. No overlap whatsoever is allowed here or we'll hit infinite recursion (existing tests caught that).